src/app/main.rs

```diff
@@ -260,38 +243,80 @@ where
-            use super::control;
+        let local_identity = match identity {
+            Conditional::None(r) => Conditional::None(r),
+            Conditional::Some((local_identity, crt_store)) => {
+                use super::control;
 
-            let tls_server_identity = config
-                .tls_settings
-                .as_ref()
-                .and_then(|s| s.controller_identity.clone().map(|id| id));
+                let id_config = match config.identity_config.as_ref() {
+                    Conditional::Some(c) => c.clone(),
+                    Conditional::None(_) => unreachable!(),
+                };
 
-            // If the controller is on localhost, use the inbound keepalive.
-            // If the controller is remote, use the outbound keepalive.
-            let keepalive = control_host_and_port.as_ref().and_then(|a| {
-                if a.is_loopback() {
+                // If the service is on localhost, use the inbound keepalive.
+                // If the service. is remote, use the outbound keepalive.
+                let keepalive = if id_config.svc.addr.is_loopback() {
                     config.inbound_connect_keepalive
                 } else {
                     config.outbound_connect_keepalive
-                }
-            });
+                };
+
+                let svc = connect::Stack::new()
+                    .push(phantom_data::layer())
+                    .push(tls::client::layer(Conditional::Some(
+                        id_config.trust_anchors.clone(),
+                    )))
+                    .push(keepalive::connect::layer(keepalive))
+                    .push(control::client::layer())
+                    .push(control::resolve::layer(dns_resolver.clone()))
+                    .push(reconnect::layer().with_fixed_backoff(config.control_backoff_delay))
+                    .push(svc::timeout::layer(config.control_connect_timeout))
+                    .push(http_metrics::layer::<_, classify::Response>(
+                        ctl_http_metrics.clone(),
+                    ))
+                    .push(proxy::grpc::req_body_as_payload::layer())
+                    .push(phantom_data::layer())
+                    .push(control::add_origin::layer())
+                    .push(buffer::layer(config.destination_concurrency_limit))
+                    .push(limit::layer(config.destination_concurrency_limit))
+                    .make(&id_config.svc)
+                    .unwrap_or_else(|e| panic!("failed to build dst_svc: {}", e));
+
+                task::spawn(
+                    identity::Daemon::new(id_config, crt_store, svc)
+                        .map_err(|_| error!("identity task failed")),
+                );
+
+                Conditional::Some(local_identity)
+            }
+        };
+
+        let dst_svc = config.destination_addr.as_ref().map(|addr| {
+            use super::control;
+
+            //let tls_server_identity = config.destination_identity.clone();
```

Comment:
Is there a reason we commented this out rather than removing it? AFAICT this is no longer necessary since it's now provided by the `ControlAddr` type?